feat: replay data services

doudou opened the pull request:
Depends on:
- [ ] https://github.com/rock-core/tools-syskit/pull/408

doudou committed:
fix: require 'interface/core' when available

This avoids loading the v1 interface when it's not meant to be.

doudou committed:
feat: add support to narrow a TaskStreams into a data service

This will allow us to replay a data service instead of a whole task,
making replaying things like devices or - generally speaking - complex
tasks simpler and more future-proof.

wvmcastro commented on test/task_streams_test.rb:
```diff
@@ -256,5 +256,75 @@ def validate_stream_name(object_name, stream_name)
+                assert_equal "task.porta", srv_streams.p_port.name
+            end
+
+            it "enumerates the mapped ports" do
```
It bothers me that the test proposes to test it enumerates portS but in the test body it is testing against only one

doudou replied:
Fixed.

wvmcastro reviewed (commented):
I think I understand what you have done here, but I can't connect how it solves the camera log replay

doudou committed:
fix: enable Syskit.conf.strict_model_for in the unit test suite

doudou committed:
fix: explicitly use syskit models in replay deployment definitions

Syskit was until now expecting an orogen model, which was then
re-resolved to a Syskit model "under the hood". This was messed
up by the fact that in syskit-log we define more than one syskit
model for a given orogen model.

Explicitly pass syskit models for clarity and comment where things
are still a bit ... shaky.

This fixes running service-narrowed task streams (the value returned by
TaskStreams#as_data_service)

doudou commented:
> I think I understand what you have done here, but I can't connect how it solves the camera log replay

One of the major current issues with replay scripts is that you have to pretend that the replayed tasks is what it was. For device drivers, it means it has to match what a device driver does, which for instance mean having a device associated with it. 

In other cases, the issue is with dynamic services. If the port was part of a dynamic service, you need to update the model to match. A pain.

This makes sure we can easily replay some ports by telling Syskit they are service X and then injecting them in compositions that accept service X. No more pretending.

doudou committed:
fix: typo in the definition of plain_task_model in the root ReplayTaskContext

doudou committed:
fix: use the new for_plain_model in tests when applicable
